### PR TITLE
Add tests, significant refactor

### PR DESCRIPTION
**Pull request by RichardLitt**
This adds tests, refactors to use promises and async, implements standards, adds a CLI file,
adds a Travis file for testing, allows you to change the destination name,
and has better logging.

### CODE REVIEW

**Comment by RichardLitt** [comment]
@simonv3 You'll need to enable Travis. I've also added Greenkeeper, might be worth enabling?

File: test/test.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  fs.unlink(path.join(__dirname, 'fixtures/RES.md'))
+})
+
+test('throws if no email provided', async t => {
+  const error = await t.throws(fn())
+  t.is(error.message, 'An email address must be provided!')
+})
+
+test('will create file', async t => {
+  const dest = path.join(__dirname, 'fixtures/RES.md')
```
Comment (simonv3) [inline test/test.js:16]: If this fails to download a file, and the test has been run before, would this still pass because the file exists?

**Review by simonv3 — APPROVED** [review]
Thank you!!

File: test/test.js (hunk 1 above)
Comment (RichardLitt) [reply test/test.js:16]: Nope. `test.after('cleanup', cb)` removes it after the test runs.

Comment (simonv3) [reply test/test.js:16]: cool!

**Pull request merged**